sycl/doc/extensions/proposed/sycl_ext_oneapi_annotated_arg.asciidoc

```diff
@@ -194,142 +195,15 @@ namespace sycl::ext::oneapi {
-
-    // Available if the operator^= is valid for objects of type T
-    T& operator^=(const T&) noexcept;
-
-    // Available if the operator&= is valid for objects of type T
-    T& operator&=(const T&) noexcept;
-
-    // Available if the operator|= is valid for objects of type T
-    T& operator|=(const T&) noexcept;
-
-    // Available if the operator<< is valid for objects of type T
-    T operator<<(const T&) noexcept;
-    const T operator<<(const T&) const noexcept;
-
-    // Available if the operator>> is valid for objects of type T
-    T operator>>(const T&) noexcept;
-    const T operator>>(const T&) const noexcept;
-
-    // Available if the operator>>= is valid for objects of type T
-    T& operator>>=(const T&) noexcept;
-
-    // Available if the operator<<= is valid for objects of type T
-    T& operator<<=(const T&) noexcept;
-
-    // Available if the operator== is valid for objects of type T
-    bool operator==(const T&) const noexcept;
-
-    // Available if the operator!= is valid for objects of type T
-    bool operator!=(const T&) const noexcept;
+    // Available if the operator[] is valid for objects of type T, return
+    // type will match the return type of T::operator[](std::ptrdiff_t)
+    /* ... */ operator[](std::ptrdiff_t idx) const noexcept;
 
-    // Available if the operator<= is valid for objects of type T
-    bool operator<=(const T&) const noexcept;
-
-    // Available if the operator>= is valid for objects of type T
-    bool operator>=(const T&) const noexcept;
-
-    // Available if the operator&& is valid for objects of type T
-    bool operator&&(const T&) const noexcept;
-
-    // Available if the operator|| is valid for objects of type T
-    bool operator||(const T&) const noexcept;
-
-    // Available if the operator++ is valid for objects of type T
-    T& operator++() noexcept;
-
-    // Available if the operator++ is valid for objects of type T
-    T operator++(int) noexcept;
-
-    // Available if the operator-- is valid for objects of type T
-    T& operator--() noexcept;
-
-    // Available if the operator-- is valid for objects of type T
-    T operator--(int) noexcept;
-
-    // Available if the operator-> is valid for objects of type T
-    T& operator->() noexcept;
-    const T& operator->() const noexcept;
-
-    // Available if the operator[] is valid for objects of type T
-    T& operator[](std::ptrdiff_t idx) noexcept;
-    const T& operator[](std::ptrdiff_t idx) const noexcept;
-
-    // Available if the operator() is valid for objects of type T
-    template<typename... Args> auto operator()(Args... args) noexcept;
-    template<typename... Args> auto operator()(Args... args) const noexcept;
+    // Available if the operator() is valid for objects of type T, return
+    // type will match the return type of
+    //   template<typename... Args> T::operator()(Args... args)
+    template<typename... Args> /* unspecified */ operator()(Args... args) noexcept;
```

Comment:
I think this is more correct because it isn't unspecified, you specify it on the preceding line.  It's just complicated so you don't want to write it here.

```suggestion
    template<typename... Args> /* ... */ operator()(Args... args) noexcept;
```